src/ColorDelta.ml

```diff
@@ -2,8 +2,14 @@ let blend color alpha = 255. +. ((color -. 255.) *. alpha)
 
 let blendSemiTransparentColor = function
   | r, g, b, alpha when alpha < 255. ->
-      (blend r alpha, blend g alpha, blend b alpha, alpha /. 255.)
-  | colors -> colors
+      let normalizedAlpha = alpha /. 255. in
+      let (r, g, b, a) = (blend r normalizedAlpha, blend g normalizedAlpha, blend b normalizedAlpha, normalizedAlpha) in
+      assert (r >= 0. && r <= 255.);
```

Comment:
I don't think we should throw exceptions here, instead of silently failing we should return max or min values

Reply:
I've changed them with `r |> min 255. |> max 0.` in the new commit. However, maybe I can just remove these assertions because I added them to test my assumption that there may be a problem.